src/level_gen.cpp

```diff
@@ -148,7 +148,9 @@ static inline Entity createAgent(Engine &ctx, const MapObject &agentInit) {
 
     ctx.get<Goal>(agent)= Goal{.position = Vector2{.x = agentInit.goalPosition.x - ctx.data().mean.x, .y = agentInit.goalPosition.y - ctx.data().mean.y}};
     populateExpertTrajectory(ctx, agent, agentInit);
-    if(!ctx.data().params.isStaticAgentControlled && (ctx.get<Goal>(agent).position - ctx.get<Trajectory>(agent_iface).positions[0]).length() < consts::staticThreshold)
+
+    auto isStatic = (ctx.get<Goal>(agent).position - ctx.get<Trajectory>(agent_iface).positions[0]).length() < consts::staticThreshold or agentInit.markAsStatic;
```

Comment:
I think we also discussed not marking an agent as static if its speed exceed a threshold like 1 m/s. Could we add that if it isn't here?

Reply:
I'm not sure I follow. If an agent cannot reach its goal, then what difference does it make what its initial velocity is?